core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -0,0 +1,213 @@
+		Map<String, Method> getterAttributes = getGetterAndIsserAttributes(methods, attributeOwner);
+		Map<String, String> setterToAttributeName = new HashMap<>();
+		for(String attributeName: setterAttributes.keySet()) {
+			setterToAttributeName.put(setterAttributes.get(attributeName).getName(), attributeName);
+		}
+		List<FrankAttribute> result = new ArrayList<>();
+		for(Method method: methods) {
+			if(setterToAttributeName.containsKey(method.getName())) {
+				String attributeName = setterToAttributeName.get(method.getName());
+				if(getterAttributes.containsKey(attributeName)) {
+					compareGetterWithSetter(method, getterAttributes.get(attributeName), attributeOwner);
+				}
+				FrankAttribute attribute = new FrankAttribute(attributeName);
+				attribute.setDescribingElement(attributeOwner);
+				documentAttribute(attribute, method, attributeOwner);
+				result.add(attribute);
+			}
+		}
+		return result;
+	}
+
+	private Map<String, Method> getGetterAndIsserAttributes(Method[] methods, FrankElement attributeOwner) {
+		Map<String, Method> getterAttributes = getAttributeToMethodMap(methods, "get");
+		Map<String, Method> isserAttributes = getAttributeToMethodMap(methods, "is");
+		for(String isserAttributeName : isserAttributes.keySet()) {
+			if(getterAttributes.containsKey(isserAttributeName)) {
+				log.warn(String.format("For FrankElement %s, attribute %s has both a getX and an isX method",
+						attributeOwner.getSimpleName(), isserAttributeName));
+			} else {
+				getterAttributes.put(isserAttributeName, isserAttributes.get(isserAttributeName));
+			}
+		}
+		return getterAttributes;
+	}
+
+	static Map<String, Method> getAttributeToMethodMap(Method[] methods, String prefix) {
+		List<Method> methodList = Arrays.asList(methods);
+		methodList = methodList.stream()
+				.filter(FrankDocModel::isGetterOrSetter)
+				.filter(m -> m.getName().startsWith(prefix) && (m.getName().length() > prefix.length()))
+				.collect(Collectors.toList());		
+		Map<String, Method> result = new HashMap<>();
+		for(Method method: methodList) {
+			String strippedName = method.getName().substring(prefix.length());
+			String attributeName = strippedName.substring(0, 1).toLowerCase() + strippedName.substring(1);
+			result.put(attributeName, method);
+		}
+		return result;
+	}
+
+	static boolean isGetterOrSetter(Method method) {
+		boolean isSetter = method.getReturnType().isPrimitive()
+				&& method.getReturnType().getName().equals("void")
+				&& (method.getParameterTypes().length == 1)
+				&& (method.getParameterTypes()[0].isPrimitive()
+						|| method.getParameterTypes()[0].getName().equals(JAVA_STRING));
+		boolean isGetter = (
+					method.getReturnType().isPrimitive()
+					|| method.getReturnType().getName().equals(JAVA_STRING)
+				) && (method.getParameterTypes().length == 0);
+		return isSetter || isGetter;
+	}
+
+	private void compareGetterWithSetter(Method setter, Method getter, FrankElement attributeOwner) {
```

Comment:
Deze method zou je kunnen hernoemen in 'checkForTypeConflict()', zodat je weet wat je kan verwachten.

Reply:
Gedaan.